geomesa-hbase/geomesa-hbase-datastore/src/main/scala/org/locationtech/geomesa/hbase/index/HBaseFeatureIndex.scala

```diff
@@ -176,8 +176,10 @@ trait HBaseFeatureIndex extends HBaseFeatureIndexType
     import org.locationtech.geomesa.index.conf.QueryHints.RichHints
 
     /** This function is used to implement custom client filters for HBase **/
-      val toFeatures = resultsToFeatures(sft, ecql, hints.getTransform)
-      ScanConfig(Nil, toFeatures)
-
+      val toFeatures = resultsToFeatures(sft, None, hints.getTransform)
+      val remoteFilters = filter.filter.map { filter =>
```

Comment:
This should use `ecql` instead of `filter.filter` - they may not be the same, `ecql` is the remaining predicate that needs to be applied, which `filter.filter` is the entire predicate, some of which may already have been handled with range planning.